subt/depth2scan.py

```diff
@@ -0,0 +1,42 @@
+"""
+  Convert depth image to "lidar" scan
+"""
+import numpy as np
+
+from osgar.node import Node
+from osgar.bus import BusShutdownException
+
+
+class DepthToScan(Node):
+    deg30 = int(30*720/270)
+
+    def __init__(self, config, bus):
+        super().__init__(config, bus)
+        bus.register("scan")
+        self.depth = None  # initialize inputs
+        self.scan = None
+
+    def update(self):
+        channel = super().update()
+        assert channel in ["depth", "scan"], channel
+
+        if channel == 'scan':
+            assert len(self.scan) == 720, len(self.scan)
+        elif channel == 'depth':
+            assert len(self.depth) == 640 * 360, len(self.depth)
+            i = (360//2)*640
+            mid_line = np.array(self.depth[i:i+640], dtype=np.dtype('H'))
+            # 60*720/270 = 160.0 ... i.e. 160 elements to be replaced
+            # 640/160 = 4.0 ... i.e. downsample by 4
+            small = mid_line[::4]
+            mask = small == 0xFFFF
+            small[mask] = 0
+            if self.scan is not None:
+                new_scan = self.scan[:360-80] + np.flip(small, axis=0).tolist() + self.scan[360+80:]
+                assert len(new_scan) == 720, len(new_scan)
+                self.publish('scan', new_scan)
```

Comment:
I am sorry but from reading this code I have no idea what it does. I guess I just might be getting lost in all the constants and also I might not have my brightest moment right know (being sick).

Reply:
OK, I will add comments. It was the first attempt to replace part of original lidar scan by data from RGBD camera and see how it matches.

Reply:
If you did change anything, it is not helping. What I am missing is some kind of conversion from the plane given by the camera to the actual scan that has equal angles for each measurement. The thing is that a pixel in the middle of the picture covers significantly different angle than a pixel at the far side - the bigger the FOV the bigger the difference. I am just guessing here but the `60` in the above comment looks like a horizontal FOV (?).

Reply:
well, this code is already obsolete (a little bit). It is "exercise" that we could somehow integrate depth data. At the moment "slope lidar" is used (`feature/virtual-follow-track` branch).
The precise angles/distances are different.

Reply:
I would fix/improve this part in following PR

Reply:
It is ok, since you are aware that the code is not correct & I understand there are other priorities.